### ISSUE TITLE
Stop loss code not implementing correctly.

### ISSUE THREAD

**Issue by Hardwyre** [askmike/gekko#1587]
Hello.  I am trying to add stop loss percentages to the CCI strat so I can prevent losses over a certain percentage.  Looking at my backtesting, I will occasionally see high single, and sometimes double-, digit losses.

I found the stop loss guide here http://gekkowarez.com/how-to-put-a-stop-loss-into-your-gekko-strategy/ and was able to put it into a copy of the CCI strat, but I think I have implemented it wrong.

First off, I get unexpected token else errors a lot, though if I remove several of the else statements, I can get it to run, but I find that Gekko then just creates a mess of buy and sell actions and changing the percentage variable does nothing.  I suspect that I do not have this put together properly.

CCI-with-stop-loss.js
```
// helpers
var _ = require('lodash');
var log = require('../core/log.js');

// let's create our own method
var method = {};

// prepare everything our method needs
method.init = function() {
  this.currentTrend;
  this.requiredHistory = this.tradingAdvisor.historySize;

  this.age = 0;
  this.trend = {
    direction: 'undefined',
    duration: 0,
    persisted: false,
    adviced: false
  };
  //In the strategy init create the stoploss variable:
  this.stop = "";
  this.stoplosspercentage = this.settings.stoploss;
  
  this.historySize = this.tradingAdvisor.historySize;
  this.ppoadv = 'none';
  this.uplevel = this.settings.thresholds.up;
  this.downlevel = this.settings.thresholds.down;
  this.persisted = this.settings.thresholds.persistence;

  // log.debug("CCI started with:\nup:\t", this.uplevel, "\ndown:\t", this.downlevel, "\npersistence:\t", this.persisted);
  // define the indicators we need
  this.addIndicator('cci', 'CCI', this.settings);
}

// what happens on every new candle?
method.update = function(candle) {
}

// for debugging purposes: log the last calculated
// EMAs and diff.
method.log = function() {
    var cci = this.indicators.cci;
    if (typeof(cci.result) == 'boolean') {
        log.debug('Insufficient data available. Age: ', cci.size, ' of ', cci.maxSize);
        log.debug('ind: ', cci.TP.result, ' ', cci.TP.age, ' ', cci.TP.depth);
        return;
    }

    log.debug('calculated CCI properties for candle:');
    log.debug('\t', 'Price:\t\t\t', this.lastPrice);
    log.debug('\t', 'CCI tp:\t', cci.tp.toFixed(8));
    log.debug('\t', 'CCI tp/n:\t', cci.TP.result.toFixed(8));
    log.debug('\t', 'CCI md:\t', cci.mean.toFixed(8));
    if (typeof(cci.result) == 'boolean' )
        log.debug('\t In sufficient data available.');
    else
        log.debug('\t', 'CCI:\t', cci.result.toFixed(2));
}

/*
 *
 */
method.check = function(candle) {

  var price = candle.close;

    this.age++;
    var cci = this.indicators.cci;

    if (typeof(cci.result) == 'number') {

        // overbought?  Includes stoploss code after overbought parentheses close 
        if (cci.result >= this.uplevel && (this.trend.persisted || this.persisted == 0) && !this.trend.adviced && this.trend.direction == 'overbought' || this.stop != "" && price<this.stop){
        
//if you want to show a stoploss being triggered in the console:
        if(this.stop != "" && price<this.stop){
            console.log("stoplosss triggered - "+ this.stop);
        }
	}
        this.direction="short";
        else if(this.trend == "long"){
            this.stop = "";
            this.trend = this.direction;
            this.advice(this.direction);
        } 
 {
            this.trend.adviced = true;
            this.trend.duration++;
            this.advice('short');
        } 
		else if (cci.result >= this.uplevel && this.trend.direction != 'overbought'){
        if(this.stop==""){
//sets up the stoploss, you should make the .2 a variable in the strategy config really
            this.stop = price-(price*this.stoplosspercentage);
            this.direction="long";
            this.trend = this.direction;
            this.advice(this.direction);
        }
    } {
            this.trend.duration = 1;
            this.trend.direction = 'overbought';
            this.trend.persisted = false;
            this.trend.adviced = false;
            if (this.persisted == 0) {
                this.trend.adviced = true;
                this.advice('short');
            }
        } else if (cci.result >= this.uplevel) {
            this.trend.duration++;
            if (this.trend.duration >= this.persisted) {
                this.trend.persisted = true;
            }
        } else if (cci.result <= this.downlevel && (this.trend.persisted || this.persisted == 0) && !this.trend.adviced && this.trend.direction == 'oversold') {
            this.trend.adviced = true;
            this.trend.duration++;
            this.advice('long');
        } else if (cci.result <= this.downlevel && this.trend.direction != 'oversold') {
            this.trend.duration = 1;
            this.trend.direction = 'oversold';
            this.trend.persisted = false;
            this.trend.adviced = false;
            if (this.persisted == 0) {
                this.trend.adviced = true;
                this.advice('long');
            }
        } else if (cci.result <= this.downlevel) {
            this.trend.duration++;
            if (this.trend.duration >= this.persisted) {
                this.trend.persisted = true;
            }
        } else {
            if( this.trend.direction != 'nodirection') {
                this.trend = {
                    direction: 'nodirection',
                    duration: 0,
                    persisted: false,
                    adviced: false
                }
            } else {
                this.trend.duration++;
            }
            this.advice();
        }

	} else {
        this.advice();
    }

    log.debug("Trend: ", this.trend.direction, " for ", this.trend.duration);
}

module.exports = method;
```

CCI-with-stop-loss.toml
```
# constant multiplier. 0.015 gets to around 70% fit
constant = 0.015

# history size, make same or smaller than history
history = 90

# stop loss variable as a percentage multiplier, e.g. 0.2, 0.5, etc
stoploss = 1

[thresholds]
up = 100
down = -100
persistence = 0
```

Here is what I end up with after commenting out the else statements it wants me to:
![image](https://user-images.githubusercontent.com/34960903/34454276-f0f5dece-ed1c-11e7-9a3b-0f1611a03904.png)

![image](https://user-images.githubusercontent.com/34960903/34454279-fbf0fbf6-ed1c-11e7-8a3e-edec86c187ad.png)

I'm brand new to JS, so the learning curve is a bit steep, but I did do some reading throughout the night here.  I'm just at a loss at this point, especially as to why it wants me to get rid of all the else statements.

I appreciate any information.

**Comment by bezelbot** [comment]
You're not supposed to comment out the else statements. Unexpected token usually means that you have an open parenthesis or curly brace somewhere that needs to be closed before you move on to the else statement. Also, each else statement needs a preceding if statement. Trace through your code carefully to make sure each open '(' and '{' has a matching ')' and '}' and make sure your else statements come after an if.

Something like this:

if (number > 0 ) {
//Do something
} 
else if (number < 0) {
//Do something else
}
else {
//Do the default action
}

**Comment by askmike** [comment]
Apologies for never folllowing up on this, just wanted to comment that proper stop loss support is coming to Gekko very soon, see here: https://forum.gekko.wizb.it/thread-57314.html
